static/app/types/onboarding.tsx

```diff
@@ -40,6 +51,14 @@ export type OnboardingTaskDescriptor = {
    * An extra component that may be rendered within the onboarding task item.
    */
   SupplementComponent?: React.ComponentType<OnboardingSupplementComponentProps>;
+  /**
+   * If a render function was provided, it will be used to render the entire card,
+   * and the card will be rendered before any other cards regardless of completion status.
+   * the render function is therefore responsible for determining the completion status
+   * of the card by returning null when it's completed. Note that this is not a functional
+   * component so you can't use react hooks in here.
```

Comment:
I would actually probably name this `renderCard`. I would then probably just say.

> This should not be given a react component.